lib/builder/target.ex

```diff
@@ -0,0 +1,91 @@
+defmodule Burrito.Builder.Target do
+  use TypedStruct
+
+  alias __MODULE__
+  alias Burrito.Builder.Log
+  alias Burrito.Util
+
+  @old_targets [:darwin, :win64, :linux, :linux_musl]
+
+  @required_key [:os, :cpu]
+
+  typedstruct do
+    field :alias, atom(), enforce: true
+    field :cpu, atom(), enforce: true
+    field :os, atom(), enforce: true
+    field :qualifiers, keyword(), enforce: true
+    field :otp_version, String.t(), enforce: true
+    field :debug?, boolean(), enforce: true
+  end
+
+  @spec init_target(keyword(), atom(), boolean()) :: Burrito.Builder.Target.t()
+  def init_target(target, target_alias, debug?) do
+    base = Keyword.take(target, @required_key)
+    extra_qualifiers = Keyword.drop(target, @required_key)
+
+    %Target{
+      alias: target_alias,
+      cpu: base[:cpu],
+      os: base[:os],
+      qualifiers: extra_qualifiers,
+      otp_version: Util.get_otp_version(),
```

Comment:
Since this field is only used when ERTS is being fetched from a URL, I wonder if this call should be moved there instead? Right now, if the `:local_erts` qualifier is included, there's a (benign) risk of this field being set to a different version than the given ERTS build.

Two other possible options worth considering are:
1) hoisting `:erts_location` up to here, and setting that using build qualifiers
2) adding `Util.get_otp_version/1`, that takes a path to an ERTS build and returns the version for that build

Option (2) may be more challenging, however, because you accept the ERTS build as a tarball, which won't have been extracted by this point.

Personally, my preference would be something along the lines of option (1), because it makes for a more declarative specification of the build target, and also gives you support for other ERTS locations for nearly free. For example, to take a page from `mix.exs`, you'd easily be able to support build targets like the following:

```elixir
[
  # Specifying OTP by version; translated to a URL behind the scenes
  {:production, ..., otp_version: "24.1.7"},
  # Inherits the OTP version from the application
  {:dev, ...},
  # Specifies an OTP build by path
  {:custom, ..., path: "path/to/local/otp/build.tar"}
  # Points to a remote OTP build (on-premise Artifactory?)
  {:remote, ..., url: "https://our-build-server/some/remote/otp/build.tar"}
]
```

This definitely isn't a massive problem or anything like that, but the possible source of data inconsistency here suggests to me that some of these concerns are happening in the wrong place.

Reply:
I've implemented an ERTS resolver that matches up with what you suggested (and what we discussed offline) -- it should allow users to specify not only URLs to ERTS builds, but also local archives of one